src/stores/callAPIMiddleware.js

```diff
@@ -0,0 +1,38 @@
+export function callAPIMiddleware({ dispatch, getState }) {
+  return (next) => (action) => {
+    const { types, doPerform } = action;
+
+    if (!types) {
+      // Normal action: pass it on
+      return next(action);
+    }
+
+    if (!types.request || !types.success || !types.error) {
+      throw new Error('One or more action types are missing.');
+    }
+
+    if (typeof doPerform !== 'function') {
+      throw new Error('Expected doPerform to be a function.');
+    }
+
+    dispatch({
+      type: types.request,
+    });
+
+    return doPerform(getState())
+      .then((response) => {
+        dispatch({
+          response,
+          type: types.success,
+        });
+      })
+      .catch((error) => {
+        dispatch({
+          error,
+          type: types.error,
+        });
+      });
+
+    return next(action);
```

Comment:
I believe this is unreachable because of returning on line `22`

Reply:
You forgot this in, the middlewares placed after this one in the list will not be called

Reply:
No it's gone in one of the later commits, possibly github was out of date. And it shouldn't be called anyways :+1:  since the action isn't a proper action and the next middlewares won't know what to do with it.